python/pyspark/ml/wrapper.py

```diff
@@ -87,9 +90,22 @@ def _new_java_array(pylist, java_class):
           - bool -> sc._gateway.jvm.java.lang.Boolean
         """
         sc = SparkContext._active_spark_context
-        java_array = sc._gateway.new_array(java_class, len(pylist))
-        for i in xrange(len(pylist)):
-            java_array[i] = pylist[i]
+        java_array = None
+        if len(pylist) > 0 and isinstance(pylist[0], list):
+            # If pylist is a 2D array, then a 2D java array will be created.
+            # The 2D array is a square, non-jagged 2D array that is big enough for all elements.
+            # Currently, this is only used by StringIndexerModel.from_arrays_of_labels
```

Comment:
I don't feel strongly about it. But only used by `StringIndexerModel.from_arrays_of_labels` seems not necessary. It can be easily out-of-date and forgotten to update.